GUACAMOLE-699: Update german translations

leetxyz opened the pull request:
Updated the main german translation and added translations for the extensions.
I also switched the translation from a non-formal ("Du") to the formal ("Sie") expression.
Please note that i can't test the extension-translations, but checked them against a JSON-Validator.

Feel free to contact me through Jira for questions / clarifications (https://issues.apache.org/jira/browse/GUACAMOLE-699)

mike-jumper commented:
Please also squash your commits if there is no logical change worth citing in the message. Repeating the same "GUACAMOLE-699: Add initial german translation" message seven times doesn't add any value to the history.

necouchman commented:
@leetxyz: Wondering if you can fix up these issues so that we can merge these changes:
- Remove translation strings that are not actually translated
- Squash commits.

leetxyz commented:
Dear all
Sorry for the delay, was on holiday.

- I squashed the commits
- I added the translations the guac-manifests

For the unneeded translations and @APP-Notations, could you please give me an example?

thanks and best regards

necouchman commented:
@mike-jumper: Not sure on some of these what should be left and what should be removed...
- Lines for empty strings (fields with no headers, empty field options, etc.)?
- Lines that refer to other lines (`@:APP.PARAMETER`, etc.)?

mike-jumper commented:
> Lines for empty strings (fields with no headers, empty field options, etc.)?

These should be kept if the empty string is the correct German translation for what is represented by an empty string in English (probably the case).

> Lines that refer to other lines (@:APP.PARAMETER, etc.)?

Same here - these should be kept if the translation as written is correct for German (assuming the line being referenced with `@:` exists in the new translation - angular-translate can't reference strings across translations).

The general rule is to omit strings which are untranslated because the translator couldn't or didn't translate them. If the string *has* been translated and just happens to be the same in both languages, then it should be duplicated. The current exceptions to this are `APP.NAME` and `APP.VERSION` which we only want defined in one location for simplicity and support for downstream branding.

necouchman commented on guacamole/src/main/webapp/translations/de.json:
```diff
@@ -400,27 +534,56 @@
+        "FIELD_OPTION_TERMINAL_TYPE_VT100"          : "vt100",
+        "FIELD_OPTION_TERMINAL_TYPE_VT220"          : "vt220",
+        "FIELD_OPTION_TERMINAL_TYPE_XTERM"          : "xterm",
+        "FIELD_OPTION_TERMINAL_TYPE_XTERM_256COLOR" : "xterm-256color",
```
Are these accurate translations?

berndhose replied:
translation is ok

necouchman commented on guacamole/src/main/webapp/translations/de.json:
```diff
@@ -400,27 +534,56 @@
         "NAME" : "SSH",
 
         "SECTION_HEADER_AUTHENTICATION" : "Authentifizierung",
+        "SECTION_HEADER_BEHAVIOR"       : "Terminal behavior",
```
Translation, or copy?

berndhose replied:
Should be "Terminalverhalten"

leetxyz replied:
Thanks, fixed.

mike-jumper commented on extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,112 @@
+
+    "PASSWORD_POLICY" : {
+
+        "ERROR_CONTAINS_USERNAME"      : "Passwörten dürfen Ihren Benutzername nicht enthalten.",
```
From @berndhose on #399:

> I have reviewed the proposed changes of #375 and they seem to be ok, apart from one occurrence of the word "Passwörten", which should read "Passwörter".

berndhose replied:
reviewed some more, see following messages

leetxyz replied:
Thanks, fixed.

berndhose commented on extensions/guacamole-auth-duo/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,13 @@
+{
+
+    "DATA_SOURCE_DUO" : {
+        "NAME" : "Duo TFA Backend"
```
translation is ok

berndhose commented on extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,112 @@
+
+        "SECTION_HEADER_CONCURRENCY"    : "Verbindungslimite",
+        "SECTION_HEADER_LOAD_BALANCING" : "Lastverteilung",
+        "SECTION_HEADER_GUACD"          : "Guacamole Proxy Parameter (guacd)"
```
translation is ok

berndhose commented on extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,112 @@
+    },
+
+    "DATA_SOURCE_MYSQL" : {
+        "NAME" : "MySQL"
```
translation is ok

berndhose commented on extensions/guacamole-auth-openid/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,12 @@
+{
+
+    "DATA_SOURCE_OPENID" : {
+        "NAME" : "OpenID SSO Backend"
```
translation is ok

berndhose commented on extensions/guacamole-auth-quickconnect/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,18 @@
+{
+
+    "DATA_SOURCE_QUICKCONNECT" : {
+        "NAME" : "QuickConnect"
```
translation is ok

berndhose commented on extensions/guacamole-auth-radius/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,13 @@
+{
+
+    "DATA_SOURCE_RADIUS" : {
+        "NAME" : "RADIUS Backend"
```
translation is ok

berndhose commented on extensions/guacamole-auth-totp/src/main/resources/translations/de.json:
```diff
@@ -0,0 +1,34 @@
+{
+
+    "DATA_SOURCE_TOTP" : {
+        "NAME" : "TOTP TFA Backend"
```
translation is ok

berndhose commented on guacamole/src/main/webapp/translations/de.json:
```diff
@@ -336,47 +437,80 @@
         "SECTION_HEADER_DEVICE_REDIRECTION" : "Geräteumleitung",
         "SECTION_HEADER_DISPLAY"            : "Bildschirm",
+        "SECTION_HEADER_GATEWAY"            : "Remote Desktop Gateway",
+        "SECTION_HEADER_LOAD_BALANCING"     : "Load Balancing",
```
translation is ok

leetxyz commented on guacamole/src/main/webapp/translations/de.json:
```diff
@@ -4,29 +4,36 @@
     "APP" : {
 
+        "NAME"    : "Apache Guacamole",
+        "VERSION" : "${project.version}",
```
Thanks, omitted.

leetxyz commented on guacamole/src/main/webapp/translations/de.json:
```diff
@@ -261,23 +292,74 @@
-        "FIELD_HEADER_CREATE_NEW_CONNECTION_GROUPS"  : "Erstelle neue Verbindungsgruppe:",
+        "FIELD_HEADER_CHANGE_OWN_PASSWORD"           : "Eigenes Passwort ändern:",
+        "FIELD_HEADER_CREATE_NEW_USERS"              : "Neuen Benutzer erstellen:",
+		"FIELD_HEADER_CREATE_NEW_USER_GROUPS"        : "Neue Benutzergruppe erstellen:",
```
Thanks, should be fixed.

leetxyz commented:
Dear @berndhose 

Thank you for your support, went through all comments and should have fixed them all and updated the PR.

mike-jumper commented:
The only issues I see are:

1. The entries within the `translations` array in `guac-manifest.json` should be maintained in alphabetical order.
2. Indentation should use spaces, not tabs (the new lines in the various `guac-manifest.json` have been indented with tabs)

Other than the above, looks good to me.

leetxyz commented:
>     1. The entries within the `translations` array in `guac-manifest.json` should be maintained in alphabetical order.

Should be fixed.

>     2. Indentation should use spaces, not tabs (the new lines in the various `guac-manifest.json` have been indented with tabs)

Should be fixed.

I also found missing de.json references in auth-jdbc and auth-ldap and added them.

mike-jumper commented:
Looks great to me.
